src/models/geograph.py

```diff
@@ -795,6 +945,9 @@ class label in `valid_classes`, as long as they are less than
                 self.barrier_classes: List[Union[str, int]] = barrier_classes
                 self.max_travel_distance: float = max_travel_distance
                 self.add_distance: bool = add_distance
+                self.metrics: Dict[str, Metric] = {}
+                self.class_metrics: Dict[str, Dict[str, Metric]] = {}
```

Comment:
I think we don't want `metrics` and `class_metrics` inside the `isinstance`, right? Because if loading from file we also want to initialise these.

Reply:
Sorry , I am not sure I fully understand  😉  Do you mean to move them to the `_load_from` methods such that they can be loaded from memory, if applicable? If yes, I think that'd be a good idea

Reply:
I think that there are situations where we might not want to load metrics, so let's leave that - but at the moment if we load from file the `metrics` and `class_metrics` dicts are not initialised so if someone calls `get_metric` after loading from file I think it won't work?

Reply:
Resolved in a2050ef